be/src/formats/parquet/file_reader.cpp

```diff
@@ -142,18 +141,17 @@ Status FileReader::_parse_footer(FileMetaData** file_metadata, int64_t* file_met
                                                    PARQUET_FOOTER_SIZE - metadata_length,
                                            &metadata_length, TProtocolType::COMPACT, &t_metadata));
     int64_t before_bytes = CurrentThread::current().get_consumed_bytes();
-    *file_metadata = new FileMetaData();
-    RETURN_IF_ERROR((*file_metadata)->init(t_metadata, _scanner_ctx->case_sensitive));
+    FileMetaData* file_metadata = new FileMetaData();
+    RETURN_IF_ERROR(file_metadata->init(t_metadata, _scanner_ctx->case_sensitive));
```

Comment:
The file meta will be a mem leak if it return with error.

Reply:
@imay thanks. yes, it's a potential bug.